azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java

```diff
@@ -272,72 +282,106 @@ public int getHttpProxyPort() {
+            return subscriptions[0].subscriptionId();
+        }
+        final String defaultId = Optional.ofNullable(az.getCurrentSubscription()).map(Subscription::subscriptionId).orElse(null);
+        final List<SubscriptionOption> wrapSubs = Arrays.stream(subscriptions).map(t -> new SubscriptionOption(t))
+                .sorted()
+                .collect(Collectors.toList());
+        final SubscriptionOption defaultValue = wrapSubs.stream()
+                .filter(t -> StringUtils.equalsIgnoreCase(t.getSubscriptionId(), defaultId)).findFirst().orElse(null);
+        final TextIO textIO = TextIoFactory.getTextIO();
+        final SubscriptionOption subscriptionOptionSelected = new CustomTextIoStringListReader<SubscriptionOption>(() -> textIO.getTextTerminal(), null)
+                .withCustomPrompt(String.format("Please choose a subscription%s: ",
+                        highlightDefaultValue(defaultValue == null ? null : defaultValue.getSubscriptionName())))
+                .withNumberedPossibleValues(wrapSubs).withDefaultValue(defaultValue).read("Available subscriptions:");
+        if (subscriptionOptionSelected == null) {
+            throw new AzureExecutionException("You must select a subscription.");
+        }
+        return subscriptionOptionSelected.getSubscription().subscriptionId();
+    }
+
+    protected Azure getOrCreateAzureClient() throws AzureAuthFailureException, AzureExecutionException {
+
         try {
-            final AzureEnvironment environment;
-            if (!AzureAuthHelper.validateEnvironment(auth.getEnvironment())) {
-                Log.prompt(String.format(UNSUPPORTED_AZURE_ENVIRONMENT, auth.getEnvironment()));
-                environment = AzureEnvironment.AZURE;
-            } else {
-                environment = AzureAuthHelper.getAzureEnvironment(auth.getEnvironment());
+            final MavenAuthConfiguration mavenAuthConfiguration = auth == null ? new MavenAuthConfiguration() : auth;
+            mavenAuthConfiguration.setType(getAuthType());
+            azureCredentialWrapper = MavenAuthUtils.login(session, settingsDecrypter, mavenAuthConfiguration);
+            if (Objects.isNull(azureCredentialWrapper)) {
+                return null;
             }
-            final String environmentName = AzureAuthHelper.getAzureEnvironmentDisplayName(environment);
-            if (environment != AzureEnvironment.AZURE) {
-                Log.prompt(String.format(USING_AZURE_ENVIRONMENT, environmentName));
+
+            final AzureEnvironment env = azureCredentialWrapper.getEnv();
+            final String environmentName = AuthHelper.azureEnvironmentToString(env);
+            if (env != AzureEnvironment.AZURE) {
+                Log.prompt(String.format(USING_AZURE_ENVIRONMENT, TextUtils.green(environmentName)));
+            }
+            Log.info(azureCredentialWrapper.getCredentialDescription());
+            final Azure tempAzure = Azure.configure()
```

Comment:
May extract codes for get default subscription to a new method, as `getOrCreateAzureClient` is a bit larger